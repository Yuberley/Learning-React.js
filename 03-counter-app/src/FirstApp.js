import React from "react";
import PropTypes from "prop-types";

const FirstApp = ({greet, subtitle}) => {

    return (
        <>
            <h1> { greet } </h1>
            <p> { subtitle }</p>
        </>
    );
}

FirstApp.propTypes = { 
    greet: PropTypes.string.isRequired
}

FirstApp.defaultProps = {
    subtitle: 'Lorem ipsum dolor sit amet, consectetur adipss'
}

export default FirstApp;