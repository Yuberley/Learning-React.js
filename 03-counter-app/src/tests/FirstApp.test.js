import React from 'react';
import { render } from "@testing-library/react";
import FirstApp from "../FirstApp";


describe('Test on file FirstApp', () => {

    // test('Should show "Hello Im Goku"', () => {

    //     const greet = 'Hello Im Goku';
        
    //     const { getByText } = render( <FirstApp greet = 'Hello Im Goku'/> );

    //     expect( getByText( greet ) ).toBeInTheDocument();

    // })

})