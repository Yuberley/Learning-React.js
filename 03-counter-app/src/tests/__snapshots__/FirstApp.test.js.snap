// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on file FirstApp should render the component <FirstApp /> successfully 1`] = `
<Fragment>
  <h1>
    Hello Im Goku
  </h1>
  <p>
    Lorem ipsum dolor sit amet, consectetur
  </p>
</Fragment>
`;
